File: frontend/user/lib/data/mapper/product/product_detail_mapper.dart
import 'package:b2205946_duonghuuluan_luanvan/data/models/product/product_detail_model.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/product/product_detail.dart';

class ProductDetailMapper {
  static ProductDetail fromModel(ProductDetailModel model) {
    return ProductDetail(
      id: model.id,
      colorId: model.colorId,
      colorName: model.colorName,
      colorHex: model.colorHex,
      sizeId: model.sizeId,
      size: model.sizeLabel,
      price: model.price,
      isActive: model.isActive,
    );
  }
}

File: backend/app/infrastructure/database/mappers/product_detail_mapper.py
from app.domain.entities.product_detail_entity import ProductDetailEntity
from app.infrastructure.database.models.product_detail import ProductDetail


class ProductDetailMapper:
    @staticmethod
    def to_entity(model: ProductDetail) -> ProductDetailEntity:
        return ProductDetailEntity(
            id=model.id,
            product_id=model.product_id,
            color_id=model.color_id,
            size_id=model.size_id,
            price=model.price,
            is_active=model.is_active,
        )

    @staticmethod
    def to_model(entity: ProductDetailEntity) -> ProductDetail:
        return ProductDetail(
            id=entity.id,
            product_id=entity.product_id,
            color_id=entity.color_id,
            size_id=entity.size_id,
            price=entity.price,
            is_active=entity.is_active,
        )
